src/daft-sql/src/modules/python.rs

```diff
@@ -9,3 +24,101 @@ impl SQLModule for SQLModulePython {
+            daft_dsl::LiteralValue::Int32(_) => unreachable_variant!(Int32),
+            daft_dsl::LiteralValue::UInt32(_) => unreachable_variant!(UInt32),
+            daft_dsl::LiteralValue::Int64(u) => u.into_py_any(py),
+            daft_dsl::LiteralValue::UInt64(_) => unreachable_variant!(UInt64),
+            daft_dsl::LiteralValue::Timestamp(..) => unreachable_variant!(Timestamp),
+            daft_dsl::LiteralValue::Date(_) => unreachable_variant!(Date),
+            daft_dsl::LiteralValue::Time(..) => unreachable_variant!(Time),
+            daft_dsl::LiteralValue::Duration(..) => unreachable_variant!(Duration),
+            daft_dsl::LiteralValue::Interval(..) => todo!(),
+            daft_dsl::LiteralValue::Float64(f) => f.into_py_any(py),
+            daft_dsl::LiteralValue::Decimal(..) => unreachable_variant!(Decimal),
+            daft_dsl::LiteralValue::Series(series) => daft_core::python::PySeries {
+                series: series.clone(),
+            }
+            .into_py_any(py),
+            daft_dsl::LiteralValue::Python(_) => unreachable_variant!(Python),
+            daft_dsl::LiteralValue::Struct(_) => todo!(),
+        },
+        _ => Err(
+            DaftError::InternalError("expected a literal, found an expression".to_string()).into(),
+        ),
+    }
+}
+
+impl SQLFunction for WrappedUDFClass {
+    fn to_expr(
+        &self,
+        inputs: &[sqlparser::ast::FunctionArg],
+        planner: &crate::SQLPlanner,
+    ) -> crate::error::SQLPlannerResult<daft_dsl::ExprRef> {
+        #[cfg(feature = "python")]
+        {
+            let e: DaftResult<PyExpr> = pyo3::Python::with_gil(|py| {
+                let mut args = Vec::with_capacity(inputs.len());
+                let kwargs = PyDict::new(py);
+
+                for input in inputs {
+                    match input {
+                        // all named (kwargs are parsed as python literals instead of exprs)
+                        sqlparser::ast::FunctionArg::Named {
+                            name,
+                            arg,
+                            operator: _,
+                        } => {
+                            let expr = planner.try_unwrap_function_arg_expr(arg)?;
+
+                            let pyany = lit_to_py_any(py, expr.as_ref())?;
+                            kwargs.set_item(name.to_string(), pyany)?;
+                        }
+                        sqlparser::ast::FunctionArg::Unnamed(arg) => {
+                            let expr = planner.try_unwrap_function_arg_expr(arg)?;
+
+                            let py_expr = PyExpr { expr };
+                            args.push(py_expr.into_expr_cls(py)?);
+                        }
+                    }
+                }
+                let args = PyTuple::new(py, &args)?;
+
+                Ok(self.call(py, args, Some(&kwargs))?)
+            });
+            let e = e?;
+
+            Ok(e.expr)
+        }
```

Comment:
It might be worth adding some rust tests for this block, hard to review since codecov has lit it up.